src/dart_document_symbol_provider.ts

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+import { TextDocument, DocumentSymbolProvider, SymbolInformation, CancellationToken, SymbolKind, Location, Uri, Range, Position } from "vscode";
+import { Analyzer, getSymbolKindForElementKind } from "./analyzer";
+import { toRange } from "./utils";
+import * as as from "./analysis_server_types";
+
+export class DartDocumentSymbolProvider implements DocumentSymbolProvider {
+	private analyzer: Analyzer;
+
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+
+	provideDocumentSymbols(document: TextDocument, token: CancellationToken): Thenable<SymbolInformation[]> {
+		let file = document.fileName;
+
+		this.analyzer.analysisSetSubscriptions({
+			subscriptions: { 'OUTLINE': [file] }
+		});
+
+		// TODO: Improve the presentation to make the file structure clearer?
```

Comment:
I removed this TODO: - looking at this, I think we're returning the right info from the POV of vscode; we'll rely on them to present things well so that user's can easily navigate the file.